### PR TITLE
Resolve Process Compose Celestia Bug

### PR DESCRIPTION
**Pull request by l-monninger**
# Summary
- **RFCs**:  $\emptyset$.
- **Categories**: `protocol-units`

Fixes Celestia orchestration bugs causing local process compose run to fail. 

# Testing

# Outstanding issues

### CODE REVIEW

**Review by musitdev — APPROVED** [review]

File: flake.nix [hunk 1]
```diff
@@ -154,6 +154,12 @@
 
               # export PKG_CONFIG_PATH=$PKG_CONFIG_PATH_FOR_TARGET
 
+              # Export linker flags if on Darwin (macOS)
+              if [[ "$(${pkgs.stdenv.hostPlatform.system})" =~ "darwin" ]]; then
+                export LDFLAGS="-L/opt/homebrew/opt/zlib/lib"
+                export CPPFLAGS="-I/opt/homebrew/opt/zlib/include"
```
Comment (mzabaluev) [inline flake.nix:160]: This does not belong in this PR.
Also, there's got to be a solution that does not result in (theoretically) non-idempotent builds.
